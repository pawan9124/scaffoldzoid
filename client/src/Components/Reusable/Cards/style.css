.card_style {
  margin: 5px;
}
.card_image {
  height: 75px !important;
  width: 75px !important;
}
a {
  text-decoration: none;
}

.card_section {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}
.card_image_section {
  flex: 1;
}
.card_details {
  flex: 2;
}
