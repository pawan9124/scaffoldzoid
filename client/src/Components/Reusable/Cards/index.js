import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import { Link } from "react-router-dom";
import Avatar from "@material-ui/core/Avatar";
import PropTypes from "prop-types";
import "./style.css";

const useStyles = makeStyles({
  root: {
    maxWidth: 300,
  },
  bullet: {
    display: "inline-block",
    margin: "0 2px",
    transform: "scale(0.8)",
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
});

export default function Cards(props) {
  const classes = useStyles();

  return (
    <Card className={`${classes.root} card_style`} data-test="cardComponent">
      <CardContent>
        <div className="card_section">
          <div className="card_image_section">
            <Link to={`/profile/${props?.user?._id}`}>
              <Avatar alt="C" src={props?.avatar} className="card_image" />
            </Link>
          </div>
          <div className="card_details">
            <Link to={`/profile/${props?.user?._id}`}>
              <div className="card_username">{props?.user?.username}</div>
            </Link>
            <div className="card_description">
              <p>{props?.description.slice(0, 75) + "..."}</p>
            </div>
          </div>
        </div>
      </CardContent>
    </Card>
  );
}
