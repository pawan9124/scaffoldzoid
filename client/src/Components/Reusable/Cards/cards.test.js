import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../../testUtils";
import Cards from "./index";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

const props = {
  user: { _id: "uijjikk#dsdf", username: "test_user" },
  avatar: "https://i.ibb.co/dkzZ2qr/selena.jpg",
  description: "This is demo test",
};

describe("Taking snap shot", () => {
  it("Taking the snapshot of the Card Component", () => {
    const tree = renderer
      .create(
        <Router>
          <Cards {...props} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Checking the render of card", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Cards {...props} />);
  });
  it("Should Render a card", () => {
    const button = findByTestAttr(wrapper, "cardComponent");
    expect(button.length).toBe(1);
  });
});

describe("Checking PropTypes", () => {
  it("Should Not throw a warning", () => {
    const expectedProps = props;

    const propsError = checkProps(Cards, expectedProps);
    expect(propsError).toBeUndefined();
  });
});
