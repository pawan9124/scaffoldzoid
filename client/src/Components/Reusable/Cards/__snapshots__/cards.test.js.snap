// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taking snap shot Taking the snapshot of the Card Component 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 card_style MuiPaper-elevation1 MuiPaper-rounded"
  data-test="cardComponent"
>
  <div
    className="MuiCardContent-root"
  >
    <div
      className="card_section"
    >
      <div
        className="card_image_section"
      >
        <a
          href="/profile/uijjikk#dsdf"
          onClick={[Function]}
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle card_image"
          >
            <img
              alt="C"
              className="MuiAvatar-img"
              src="https://i.ibb.co/dkzZ2qr/selena.jpg"
            />
          </div>
        </a>
      </div>
      <div
        className="card_details"
      >
        <a
          href="/profile/uijjikk#dsdf"
          onClick={[Function]}
        >
          <div
            className="card_username"
          >
            test_user
          </div>
        </a>
        <div
          className="card_description"
        >
          <p>
            This is demo test...
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
